algorithm_rgb.py

```diff
@@ -48,15 +48,122 @@
 
 
 # Entry point for plot-level RBG algorithm
+
+
+def excess_greenness_index(pxarray: np.ndarray):
+    """
+    Minimizes variation between different illuminations and enhances detection
+    of plants
+    """
+    red, green, blue = calculate(pxarray)
+
+    return 2 * green - (red + blue)
+
+
+def green_leaf_index(pxarray: np.ndarray):
+    """
+    Calculates the green leaf index of an image
+    """
+    red, green, blue = calculate(pxarray)
+
+    return (2 * green - red - blue) / (2 * green + red + blue)
+
+
+def cive(pxarray: np.ndarray):
+    """
+    Can measure crop growth status
+    """
+    red, green, blue = calculate(pxarray)
+
+    return 0.441 * red - 0.811 * green + 0.385 * blue + 18.78745
+
+
+def normalized_difference_index(pxarray: np.ndarray):
+    """
+    Calculates the normalized difference index of an image
+    """
+    red, green, blue = calculate(pxarray)
+
+    return 128 * ((green - red) / (green + red)) + 1
+
+
+def excess_red(pxarray: np.ndarray):
+    """
+    Finds potential illumination issues that make it difficult to
+    tease apart redness from crops/leaves from soil or camera
+    artifacts
+    """
+    red, green, blue = calculate(pxarray)
+
+    return 1.3 * red - green
+
+
+def exgr(pxarray: np.ndarray):
+    """
+    Minimizes the variation between different illuminations
+    """
+
+    return excess_greenness_index(pxarray) - excess_red(pxarray)
+
+
+def combined_indices_1(pxarray: np.ndarray):
+    """
+    Combined indices calculation 1
+    """
+
+    return excess_greenness_index(pxarray) + cive(pxarray)
+
+
+def combined_indices_2(pxarray: np.ndarray):
+    """
+    Combined indices calculation 2
+    """
+
+    return 0.36 * excess_greenness_index(pxarray) + 0.47 * cive(pxarray) + 0.17 * vegetative_index(pxarray)
+
+
+def vegetative_index(pxarray: np.ndarray):
+    """
+    Minimize illumination differences between images
+    """
+    red, green, blue = calculate(pxarray)
+
+    return green / ((red ** 0.667) * (blue ** .333))
+
+
+def ngrdi(pxarray: np.ndarray):
+    """
+    Can measure crop growth status
+    """
+    red, green, blue = calculate(pxarray)
+
+    return (green - red) / (green + red)
+
+
+def percent_green(pxarray: np.ndarray) -> float:
+    """
+    Returns the percentage of an image that is green, which can be used
+    to identify plant cover
+    """
+    red, green, blue = calculate(pxarray)
+
+    return green / (red + green + blue)
+
+
 def calculate(pxarray: np.ndarray):
```

Comment:
You are using this incorrectly. Please look at the HOWTO

Reply:
Didn't realize I still had the "--> float" in that function. I deleted it

Reply:
I was going off of lines 41-44 in the HOW_TO

Reply:
You can replace the "-> float" with "-> tuple".

Reply:
That's what you want the algorithm to return as the results of the calculations: the averages of the red, green, and blue channels?

Reply:
I asked David but he hasn't responded yet. Based on what the function looks like in the google doc, I would assume that it is the entire amount of each type of pixel rather than the averages. I can change it to that for now and revert if he says otherwise

Reply:
You need this function to return the results of the calculations, not the averages of each color channel.

Reply:
As in all of the calculations would be returned by the one function call? I'll do that